### PR TITLE
Support useWallclockAsTimestamps AVFormat options

### LINKED ISSUE #429: tuning for Zero Latency LiveStream
Hi,
I want to show a livestream from an IP-Camera with minimum latency.
Without modification my latency was ~1200ms. I read and tweaked some time around
and with the code below I have reached 440 - 560 ms latency.
The ONVIF Device Manager (which is also using ffmpeg) has on the same setup a latency of ~320ms.
So there is still space for improvement.

I think this is a very common requirement for an app, that is why 
I would like to do an community approach. 
In the code below you can see how I did it, and what I have already tried.
If you have some idea what I should try please tell me. I will try them out and I will post the optimal solution here.
I think it would be great feature if such tuning would be integrated in QtAV.

sincerely,

Christian

```Cpp
// call this before loading the stream
static void tuneForZeroLatencyLiveStream(QtAV::AVPlayer* player)
{
  QVariantHash settings;

  // sets av_opt_set_xx and av_dict_set settings
  QVariantHash avFormat;

  {// these settings seem to have no impact on latency?
    //avFormat["fpsprobesize"] = 32;
    //avFormat["framerate"] = 100; // from: https://github.com/wang-bin/QtAV/issues/369
    //avFormat["rtbufsize"] = 512 * 1024;
    //avFormat["flush_packets"] = 1;
    //avFormat["avioflags"] = "direct";
    //avFormat["max_delay"] = 1000;
    //avFormat["fflags"] = "nobuffer";
    //avFormat["vprofile"] = "baseline";
  }

  avFormat["probesize"] = 4096; // great impact but how to calc the optimum?  // from: https://github.com/wang-bin/QtAV/wiki/FFmpeg-dict-opt-in-QtAV
  avFormat["tune"] = "zerolatency,fastdecode";
  {
    // is what zerolatency does:
    //  avFormat["bframes"] = 0;
    //  avFormat["force-cfr"] = 1;
    //  avFormat["no-mbtree"] = 1;
    //  avFormat["sync-lookahead"] = 0;
    //  avFormat["sliced-threads"] = 1;
    //  avFormat["rc-lookahead"] = 0;
  }

  settings["avformat"] = avFormat;
  player->setOptionsForFormat(avFormat);
}
```

### PR DESCRIPTION
**Pull request by cfsghost**
fixed #429 

it has high latency because media source has no timestamp information(mjpeg format, lot of IP Camera and live streaming is using mjpeg), so use_wallclock_as_timestamps can solve this problem.

### CODE REVIEW

File: qml/QmlAVPlayer.cpp [hunk 1]
```diff
@@ -250,6 +251,26 @@ void QmlAVPlayer::setVideoCodecOptions(const QVariantMap &value)
...
+	return mUseWallclockAsTimestamps;
+}
+
+void QmlAVPlayer::setWallclockAsTimestamps(bool o)
+{
+    if (mUseWallclockAsTimestamps == o)
+        return;
+    mUseWallclockAsTimestamps = o;
+
+	QVariantHash opt;
```
Comment (wang-bin) [inline qml/QmlAVPlayer.cpp]: QVariant opt = mpPlayer->optionsForFormat()

File: qml/QmlAV/QmlAVPlayer.h [hunk 2]
```diff
@@ -257,6 +262,7 @@ private Q_SLOTS:
 private:
     Q_DISABLE_COPY(QmlAVPlayer)
 
+	bool mUseWallclockAsTimestamps;
```
Comment (wang-bin) [inline qml/QmlAV/QmlAVPlayer.h]: use white spaces instead of tab

**Comment by wang-bin** [comment]
player->setClockType(AVClock::VideoClock) is another solution for streams have no timestamp.
the 2nd commit is not necessary. commit 1 and 3 can be merge into 1 commit.

**Comment by cfsghost** [comment]
I've merged commits and followed your instruction to clean up code. Please check! :-)

**Comment by wang-bin** [comment]
Don't include my commits in your pull request. You can create a new branch from master and change on that branch.

**Comment by cfsghost** [comment]
my bad. i closed this then make a new PR.

**Pull request closed without merging**